builder/dockerfile/dispatchers.go

```diff
@@ -71,6 +71,10 @@ func env(b *Builder, args []string, attributes map[string]bool, original string)
 			return errBlankCommandNames("ENV")
 		}
 
+		if runtime.GOOS == "windows" {
+			// Case insensitive environment variables on Windows
+			args[j] = strings.ToUpper(args[j])
```

Comment:
@tonistiigi I don't have a strong enough opinion to justify it either way, but updated to just do the comparison.